src/ast/node.cr

```diff
@@ -26,9 +26,20 @@ module Mint
           start[0] <= line <= end[0]
         end
 
+        # start = {2,2}
+        # end = {4,3}
+        # line: 2, column: 4
         def contains?(line : Int, column : Int)
-          (start[0] <= line <= end[0]) &&
-            (start[1] <= column <= end[1])
+          case
+          when start[0] == line && end[0] == line # If on the only line
+            start[1] <= column <= end[1]
+          when start[0] == line # If on the first line
+            column > start[1]
+          when end[0] == line # If on the last line
+            column < end[1]
+          else
+            contains?(line)
+          end
```

Comment:
This should be reverted I guess.

Reply:
I remembered why this is needed.

Let's say you have a node which starts at `{2,2}` and ends at `{4,3}`, and we are testing if `{3, 10}` is included in there, the previous method will fail since: 

```
(2 <= 3 <= 4) &&   # true
(2 <= 10 <= 3)      # false 
```

But since the entire 3rth line is contained we need to change the condition so:

- if it's on the same line we can use the same condition as before
- if it's on the first line we test if it's after the start character
- if it's on the last line we test if it's before the end character
- otherwise we check if it contains the given line

Reply:
That's a valid point, thanks for the explanation!
Specs covering these cases would be quite helpful :)